macosx/Migrator.m

```diff
@@ -0,0 +1,53 @@
+//
+//  Migrator.m
+//  Celestia
+//
+//  Created by 李林峰 on 2019/10/7.
+//
+
+#import "Migrator.h"
+#include <pwd.h>
+
+@implementation Migrator
+
++ (NSInteger)supportedDataBaseVersion {
+    return 1;
+}
+
++ (NSInteger)dataBaseVersion {
+    return [[NSUserDefaults standardUserDefaults] integerForKey:@"dbVersion"];
+}
+
++ (void)tryToMigrate {
+    if ([self dataBaseVersion] != [self supportedDataBaseVersion]) {
+        [self migrateDataBaseFrom0To1];
+    }
+}
+
++ (void)migrateDataBaseFrom0To1 {
+    if ([self dataBaseVersion] != 0)
+        return;
+
+    // get the real home path
+    const char *home = getpwuid(getuid())->pw_dir;
```

Comment:
we have HomeDir() here https://github.com/CelestiaProject/Celestia/blob/1.6.x/src/celutil/unixdirectory.cpp#L144

Reply:
@375gnu HomeDir will return the result of getenv("HOME"), this is the sandboxed home path (which should be ~/Library/Containers/space.celestia.Celestia/Data). Since 1.6.1 is non sandboxed, the preference file is located in the non-sandboxed home path(~), the only way to get this seems to be using getpwuid(getuid())